### PR TITLE
Append new validations hostname and fqdn

### PR DESCRIPTION
**Pull request by senuphtyz**
Fixes Or Enhances # .

**Make sure that you've checked the boxes below before you submit PR:**
- [x] Tests exist or have been written that cover this particular change.

Change Details:

added hostname & fqdn validations

@go-playground/admins

added hostname & fqdn validations

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -794,6 +794,18 @@ Note: See Go's ParseMAC for accepted formats and types:
 
 	http://golang.org/src/net/mac.go?s=866:918#L29
 
+Hostname
+
+This validates that a stringa value is a valid Hostname
```
Comment (joeybloggs) [inline doc.go]: Just minor but if can change `stringa`to `string`

File: regexes.go [hunk 2]
```diff
@@ -60,4 +61,5 @@ var (
 	latitudeRegex            = regexp.MustCompile(latitudeRegexString)
 	longitudeRegex           = regexp.MustCompile(longitudeRegexString)
 	sSNRegex                 = regexp.MustCompile(sSNRegexString)
+	hostnameRegex            = regexp.MustCompile(hostnameRegexString)
```
Comment (joeybloggs) [inline regexes.go]: Also minor but if you could make the groups non-capturing that'd be great! eg. `(?:`

File: regexes.go [hunk 3]
```diff
@@ -30,6 +30,7 @@ const (
 	latitudeRegexString            = "^[-+]?([1-8]?\\d(\\.\\d+)?|90(\\.0+)?)$"
 	longitudeRegexString           = "^[-+]?(180(\\.0+)?|((1[0-7]\\d)|([1-9]?\\d))(\\.\\d+)?)$"
 	sSNRegexString                 = `^\d{3}[- ]?\d{2}[- ]?\d{4}$`
+	hostnameRegexString            = `^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$`
```
Comment (joeybloggs) [inline regexes.go]: Also minor but if you could make the groups non-capturing that'd be great! eg. (?:

**Comment by senuphtyz** [comment]
That validator is a great help and code hygiene is very important also to me.
Changed code according to your review.

**Pull request merged**